### ISSUE TITLE
Compile Score Timeline has unrelated files listed

### ISSUE THREAD

**Issue by missmah** [Viladoman/CompileScore#25]
![image](https://user-images.githubusercontent.com/10825214/174676579-02f805f4-5faf-4688-be67-bc3e88bff76a.png)

The files listed in the timeline aren't related to or included by each other -- it seems like maybe the database got corrupted, or somehow a pointer to a data structure is pointing to the wrong element.

Here, we see one cpp file ostensibly including a couple of totally separate CPP files (which in reality are not included by each other), and eventually building some bullet code -- which, again, is unrelated to everything above it in this hierarchy.

**Comment by missmah** [comment]
More information about this bug. I restarted `CompileScore` and reloaded the file, and the data I'm seeing is different (and consistent/sane again).

I had been clicking `Actions -> Reload` previously, which worked like a charm on Friday night -- but it seems to have failed me just now today.

**Comment by Viladoman** [comment]
This is really concerning! 
If I understand properly, same recorded data looked wrong and then right again? meaning it is an issue with the viewer and not the .scor generator?

**Comment by missmah** [comment]
Yeah, seems to be a viewer issue, and it seemed like it was showing a garbage interpretation of the data after `Reload`, and then I quit and re-launched, and everything looked correct. But `Reload` had worked fine hundreds of times before that.

I did put the machine to sleep over the weekend; but, it also has ECC memory, so I wouldn't expect it to be due to a cosmic ray style flipped bit...

**Comment by Viladoman** [comment]
I am thinking that maybe the timeline logic did not properly clear the previous tree and merged 2 different timelines together. 
I think cosmic rays is less probable than a human error in this case :D. I will have a look. 
Thanks for letting me know.
